src/workspace.coffee

```diff
@@ -383,8 +391,7 @@ class Workspace extends Model
   #
   # Returns a {Disposable} on which `.dispose()` can be called to unsubscribe.
   onDidAddTextEditor: (callback) ->
-    @onDidAddPaneItem ({item, pane, index}) ->
-      callback({textEditor: item, pane, index}) if item instanceof TextEditor
+    @emitter.on 'did-add-text-editor', callback
```

Comment:
I figured we should avoid a duplicate `instanceof TextEditor` check, so I moved the emission of the `did-add-text-editor` event into the callback above.